### PR TITLE
Fix file names for job application downloads

### PR DESCRIPTION
**Pull request by drewpotter**
Here is the initial work towards fixing the filename length within the zipped job applications downloads.

### CODE REVIEW

File: lib/download/generate_zip.rb [hunk 1]
```diff
@@ -28,18 +34,19 @@ def all_applications_generate_zip form_id
     form.submissions.where(is_submitted: true).each do |submission|
       candidate_path = "#{submission.name}".parameterize.underscore
       all_submissions_path = "all_submissions"
+      vacancy_label = form.vacancy.label.scan(/\(([^)]+)\)/).first.first
```
Comment (Levia) [inline lib/download/generate_zip.rb]: That regular expression is probably unecessarily too complex. You could simplify it like this for example: `\((.*)\)`
`.` stands for `any character` and `*` means `0 or multiple times`; so it translates to `anything`

File: lib/download/generate_zip.rb [hunk 2]
```diff
@@ -28,18 +34,19 @@ def all_applications_generate_zip form_id
...
 
       system("mkdir #{zipped_files_path}/#{candidate_path}", chdir: @path)
 
-      system("cp #{submission.cv.path} #{zipped_files_path}/#{candidate_path}/#{submission.name.parameterize.underscore}_#{submission.cv_file_name}", chdir: @path)
-      system("cp #{submission.application_form.path} #{zipped_files_path}/#{candidate_path}/#{submission.name.parameterize.underscore}_#{submission.application_form_file_name}", chdir: @path)
-      system("cp #{submission.cover_letter.path} #{zipped_files_path}/#{candidate_path}/#{submission.name.parameterize.underscore}_#{submission.cover_letter_file_name}", chdir: @path)
+      system("cp #{submission.cv.path} #{zipped_files_path}/#{candidate_path}/#{vacancy_label}_#{candidate_path}_CV#{File.extname(submission.cv_file_name)}", chdir: @path)
```
Comment (Levia) [inline lib/download/generate_zip.rb]: Same thing as above to shorten these lines of code.

**Comment by Levia** [comment]
I've noticed that the downloaded zip file name is still very long. We should probably shorten that as well.

**Comment by drewpotter** [comment]
I have refactored the code @Levia and I hope this is much better now.

**Pull request merged**